Assets/MRTK/SDK/Features/Utilities/Solvers/HandConstraintPalmUp.cs

```diff
@@ -163,5 +211,193 @@ protected override bool IsValidController(IMixedRealityController controller)
+        /// reattach to the hand.
+        /// </summary>
+        public void StartWorldLockReattachCheckCorotine()
+        {
+            StartCoroutine(WorldLockedReattachCheck());
+        }
+
+        /// <summary>
+        /// This function attenpts to generate a hand plane based on the wrist, index knuckle and pinky knuckle joints present in the hand.
+        /// On a success, it then calls GenerateActivationPoint to try to generate a hand-based activation point that the user
+        /// needs to gaze at to activate the constrained object.
+        /// On a failure, it assigns them to be default values and then returns false
+        /// </summary>
+        /// <param name="jointedHand"></param>
+        /// <param name="handPlane"></param>
+        /// <param name="activationPoint"></param>
+        /// <returns></returns>
+        private bool GenerateHandPlaneAndActivationPoint(IMixedRealityHand jointedHand, out Plane handPlane, out Vector3 activationPoint)
+        {
+            // Generate the hand plane that we're using to generate a distance value.
+            // This is done by using the index knuckle, pinky knuckle, and wrist
+            MixedRealityPose indexKnuckle;
+            MixedRealityPose pinkyKnuckle;
+            MixedRealityPose wrist;
+
+            if (jointedHand.TryGetJoint(TrackedHandJoint.IndexKnuckle, out indexKnuckle) &&
+                jointedHand.TryGetJoint(TrackedHandJoint.PinkyKnuckle, out pinkyKnuckle) &&
+                jointedHand.TryGetJoint(TrackedHandJoint.Wrist, out wrist))
+            {
+                handPlane = new Plane(indexKnuckle.Position, pinkyKnuckle.Position, wrist.Position);
+                activationPoint = handPlane.ClosestPointOnPlane(GenerateActivationPoint(jointedHand));
+                return true;
+            }
+            else // Otherwise, set the activation point and plane to default values and return false
+            {
+                handPlane = new Plane();
+                activationPoint = Vector3.zero;
+                return false;
+            }
+        }
+
+        /// <summary>
+        /// This function attempts to generate an activation point based on what the currently-selected safe zone is.
+        /// The activation point is a Vector3 that represents the "Center" point of the area that a user needs to gaze at to
+        /// activate the attached 
+        /// </summary>
+        /// <param name="jointedHand"></param>
+        /// <returns></returns>
+        private Vector3 GenerateActivationPoint(IMixedRealityHand jointedHand)
+        {
+            MixedRealityPose referenceJoint1;
+            MixedRealityPose referenceJoint2;
+
+            switch (SafeZone)
+            {
+                case SolverSafeZone.AboveFingerTips:
+                    if (!jointedHand.TryGetJoint(TrackedHandJoint.MiddleTip, out referenceJoint1) ||
```

Comment:
I'm confused about this case. What happens if solver zone is above finger tips exactly?